biz.aQute.resolve/src/biz/aQute/resolve/RunResolution.java

```diff
@@ -354,4 +390,122 @@ public String report(boolean optionals) {
 	public Map<Resource, List<Wire>> getRequired() {
 		return required;
 	}
+
+	public static class CacheDTO {
+		public String					checksum;
+		public List<VersionedClause>	runbundles;
+	}
+
+	/**
+	 * Cache this resolution for the given project. A cache can be retrieved
+	 * with getCached(Project project). This requires a real resolution
+	 */
+
+	public void cache() {
+
+		assert isOK() : "can only be called for a real resolution";
+
+		try {
+			File f = getCacheFile(project);
+
+			CacheDTO dto = new CacheDTO();
+			dto.checksum = project.getChecksum();
+			dto.runbundles = getRunBundles();
+
+			Path tmp = Files.createTempFile(f.getParentFile()
+				.toPath(), project.getName(), ".json");
```

Comment:
This gave an error in windows (see azure pipeline failure):

```
biz.aQute.resolve.RunResolutionTest > testLaunchWithBeforeLaunchResolve FAILED
    java.nio.file.InvalidPathException: Illegal char <:> at index 71: C:\Users\VssAdministrator\.bnd\default-ws\cnf\cache\4.4.0\resolutions\D:\a\1\s\biz.aQute.resolve\testdata\enroute
        at sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
        at sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
        at sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
        at sun.nio.fs.WindowsPath.parse(WindowsPath.java:94)
        at sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:255)
        at java.io.File.toPath(File.java:2234)
        at biz.aQute.resolve.RunResolution.cache(RunResolution.java:416)
        at biz.aQute.resolve.RunResolution.lambda$getRunBundles$4(RunResolution.java:469)
        at aQute.bnd.service.result.Ok.then(Ok.java:113)
        at biz.aQute.resolve.RunResolution.getRunBundles(RunResolution.java:467)
        at biz.aQute.resolve.Bndrun.getRunbundles(Bndrun.java:154)
        at aQute.bnd.build.ProjectLauncher.updateFromProject(ProjectLauncher.java:123)
        at aQute.launcher.plugin.ProjectLauncherImpl.updateFromProject(ProjectLauncherImpl.java:92)
        at aQute.bnd.build.Project.getProjectLauncher(Project.java:2818)
        at biz.aQute.resolve.RunResolutionTest.testLaunchWithBeforeLaunchResolve(RunResolutionTest.java:113)
```

Reply:
https://dev.azure.com/bndtools/bnd/_build/results?buildId=1843&view=logs&j=5d1700ad-7d0a-54b3-d968-60bb514cf4d6&t=f44c40b5-a7fb-58cc-8f9d-bc280c1c5d20&l=395